src/app/static/src/app/components/downloadResults/DownloadResults.vue

```diff
@@ -2,28 +2,58 @@
     <div class="container">
         <div class="row">
             <div class="col-sm">
-                <h4 v-translate="'exportOutputs'"></h4>
-                <a class="btn btn-red btn-lg my-3" :href=spectrumUrl>
-                    <span v-translate="'export'"></span>
-                    <download-icon size="20" class="icon ml-2" style="margin-top: -4px;"></download-icon>
-                </a>
-                <h4 class="mt-4" v-translate="'downloadCoarseOutput'"></h4>
-                <a class="btn btn-red btn-lg my-3" :href=coarseOutputUrl>
-                    <span v-translate="'download'"></span>
-                    <download-icon size="20" class="icon ml-2" style="margin-top: -4px;"></download-icon>
-                </a>
-                <h4 class="mt-4" v-translate="'downloadSummaryReport'"></h4>
-                <a class="btn btn-red btn-lg my-3" :href=summaryReportUrl>
-                    <span v-translate="'download'"></span>
-                    <download-icon size="20" class="icon ml-2" style="margin-top: -4px;"></download-icon>
-                </a>
+                <div id="spectrum-download">
+                    <h4 v-translate="'exportOutputs'"></h4>
+                    <button class="btn btn-red btn-lg my-3"
+                            @click="downloadSpectrum">
+                        <span v-translate="'export'"></span>
+                        <download-icon size="20" class="icon ml-2" style="margin-top: -4px;"></download-icon>
+                    </button>
+                    <div v-if="!uploadModalOpen">
+                        <download-progress id="spectrum-progress"
+                                           :complete="spectrum.complete"
+                                           :downloading="spectrum.downloading">
+                        </download-progress>
+                    </div>
+                    <error-alert id="spectrum-error" v-if="spectrum.error" :error="spectrum.error"></error-alert>
+                </div>
+                <div id="coarse-output-download">
+                    <h4 class="mt-4" v-translate="'downloadCoarseOutput'"></h4>
+                    <button class="btn btn-red btn-lg my-3"
+                            @click="downloadCoarseOutput">
+                        <span v-translate="'download'"></span>
+                        <download-icon size="20" class="icon ml-2" style="margin-top: -4px;"></download-icon>
+                    </button>
+                    <div v-if="!uploadModalOpen">
```

Comment:
To avoid the complexities of co-ordinating download and upload UI it might be better just to disable the button that opens the modal if a download is currently in progress?